snapcraft/cli/env.py

```diff
@@ -52,7 +52,7 @@ def __init__(
         :param str additional_providers: Additional providers allowed in the
                                          environment.
         """
-        valid_providers = ["host", "lxd"]
+        valid_providers = ["host", "multipass", "lxd", "managed-host"]
```

Comment:
Let's make it an enum. I started with managed but managed-host felt a bit better, host should go away and probably be renamed to destructive-host. Since we had not agreed on a name for it yet I had not dived into changing it.

Reply:
I will propose a follow-up PR for this one.

Reply:
issue taken care of